Added a filterQuality parameter to texture(Poor video scaling quality #53080)

zljj0818 opened the pull request:
## Description
Use bilinear default instead of nearest filter to draw texture

## Related Issues
https://github.com/flutter/flutter/issues/53080

## Tests

I added the following tests:

*Replace this with a list of the tests that you added as part of this PR. A change in behaviour with no test covering it
will likely get reverted accidentally sooner or later. PRs must include tests for all changed/updated/fixed behaviors. See [Test Coverage].*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

<!-- Links -->
[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

zljj0818 committed:
Added a filterQuality parameter to texture

engine-flutter-autoroll committed:
c7f972552 Roll Fuchsia Mac SDK from Xqev9... to uCq3V... (flutter/engine#18881) (#58952)

gspencergoog committed:
Turn off flaky indicator for flutter_gallery__back_button_memory and flutter_gallery__memory_nav (#58838)

After updating the GBoard on the devicelab machines, these tests are no longer flaky.

Commit:
[flutter_tools] rename library to be less absurd (#58891)

Instead of targets/dart.dart -> targets/common.dart

Commit:
[flutter_tools] only restrict devices based on arch + buildMode, not emulator status (#58887)

instead of restricting profile/release mode based on whether the tool thinks the device is an emulator, restrict based on the device target architecture and the requested build mode. Notably, this enables release mode on x86_64 Android emulators, but not x86 emulators since we do not support that as an AOT target.

This does not add release mode support for simulators, since this requires us to build and upload artifacts for simulator/x86_64

Commit:
[flutter_tools] fix capitalization in build commands (#58842)

LongCatIsLooong committed:
Add 2 new keyboard types and infer keyboardType from autofill hints (#56641)

HansMuller committed:
Restore some typography tests (#58843)

helin24 committed:
Send text error in JSON and print in tools (#58994)

* Revert "Revert "Send text error in JSON and print in tools (#58284)" (#58872)"

This reverts commit c2d5e18cb2df9400ab3b34f12fb1eba957084e2f.

* Put streamListen in try/catch if extension events already listened for

jmagman committed:
Support work profiles and multiple Android users for run, install, attach, drive (#58815)

Commit:
fix analysis on master (#59001)

Commit:
Revert "Send text error in JSON and print in tools (#58994)" (#59002)

This reverts commit f0174b176a87ca12a8128ca1cd5edadcfede46b5.

Commit:
[flutter_tools] change service worker load to NOTICES (#58890)

update flutter_service_worker.js to NOTICES

dev-aentgs committed:
Update animation.dart (#58971)

goderbauer commented:
Looks like this is the framework counterpart to https://github.com/flutter/engine/pull/18814? /cc @liyuqian

This PR is also missing tests. Can you please add some?

goderbauer commented on packages/flutter/lib/src/rendering/layer.dart:
```diff
@@ -603,6 +604,12 @@ class TextureLayer extends Layer {
   /// un-freezes it when it is certain that a frame with the new size is ready.
   final bool freeze;
 
+  /// Used to set the filterQuality of the texture
```
please use full sentences in doc comments. They should end all with a `.`.

Also: The first sentence of a doc should be its own paragraph summarizing what's documented.

perclasson committed:
Add collapsed height param to SliverAppBar (#58593)

engine-flutter-autoroll committed:
Roll Engine from c7f9725521c0 to f581f428e981 (5 revisions) (#59007)

jmagman committed:
Build iOS apps using Swift Packages (#59009)

gspencergoog committed:
Make _RenderSlider not be a semantics container (#58535)

This PR makes _RenderSlider not be a semantics container. This is so that the FocusableActionDetector in the Slider widget will get to aggregate the semantics information, since otherwise Talkback won't focus the slider because it thinks that the focus node doesn't have anything to say (which it doesn't but the _RenderSlider child does). If the _RenderSlider is a semantics container, then it keeps its speakable information to itself, but it isn't the Focus widget, so when the keyboard focus goes to the focus node, the accessibility focus doesn't move.

Since the _RenderSlider is always wrapped by the Slider widget, there's nothing lost in making it not be a container.

CareF committed:
fix typo in macrobenchmarks/lib/main (#58783)

jmagman committed:
Revert "Build iOS apps using Swift Packages (#59009)" (#59025)

This reverts commit 43e28084fcdf992fc503f855addfcde30f3f3360.

chunhtai committed:
enable Navigator.of to accept a navigator element and return its stat… (#58259)

chunhtai committed:
Introduce inherited navigator observer and refactor hero controller (#58808)

engine-flutter-autoroll committed:
Roll Engine from f581f428e981 to 5edd7666287c (2 revisions) (#59011)

liyuqian committed:
Make non-flaky tests as such (#59013)

xster committed:
add a help link to the default module template readme (#59023)

iskakaushik committed:
Timeline summary contains CPU, GPU and Memory usage (#58820)

As of flutter.dev/go/engine-cpu-profiling, we collect the CPU and
Memory usage. With work being done to collect GPU usage on iOS as well.

This adds them to the timeline summary.

Fixes: https://github.com/flutter/flutter/issues/58803

jmagman committed:
Release cache lock for commands after required artifacts are downloaded (#59012)

Commit:
[flutter_tools] use correct sdk path for analysis (#58871)

flutter analyze was unconditionally using the cached dart-sdk for analysis, and was not running with the built SDK during local engine. This broke when trying to update the analyzer for null safety, since it required us to wait for the dart change to roll into the framework first.

engine-flutter-autoroll committed:
Roll Engine from 5edd7666287c to 580ab9c3c592 (3 revisions) (#59028)

Commit:
Revert "[flutter_tools] use correct sdk path for analysis (#58871)" (#59035)

This reverts commit 9ed3e5bea6c36d4b4a25ebed26e4db8ebeee04f8.

zljj0818 committed:
add more comment

zljj0818 commented:
> Looks like this is the framework counterpart to [flutter/engine#18814](https://github.com/flutter/engine/pull/18814)? /cc @liyuqian
> 
> This PR is also missing tests. Can you please add some?

Which kind of test to add? Is there any examples?

Commit:
[flutter_tools] Reland: use correct sdk path for analysis (#59081)

flutter analyze was unconditionally using the cached dart-sdk for analysis, and was not running with the built SDK during local engine. This broke when trying to update the analyzer for null safety, since it required us to wait for the dart change to roll into the framework first.

guidezpl committed:
Scale input decorator label width  (#59010)

jason-simmons committed:
Handle selection ranges where getBoxesForSelection returns an empty list (#59014)

chunhtai committed:
fix overscroll position if there is sliver before center sliver in custom scroll view (#59015)

Commit:
[flutter_tools] support bundle-sksl-path on all desktop and mobile targets (#58879)

Adds support for --bundle-sksl-path to windows, linux, macOS, and iOS.

Fixes #53115

helin24 committed:
Send text error in JSON and print in tools (#59018)

christopherfujino committed:
[flutter_tools] Fix slow ios_device_start_prebuilt_test (#59026)

engine-flutter-autoroll committed:
Roll Engine from 580ab9c3c592 to 6fd335671508 (3 revisions) (#59039)

Commit:
[flutter_tools] include dart-defines in cached kernel name (#59083)

This prevents using a cached kernel file with different defines, since --initialize-from-dill does not handle this correctly.

Fixes #58976

Commit:
[flutter_tools] support flutter run -d edge (#56173)

Support Microsoft Edge as a device for flutter run when web is enabled. Currently this only works on Windows, and does not include a validator for edge.

Fixes #55322

a14n committed:
nnbd annotation in flutter_goldens_client (#59069)

jmagman committed:
Add per-test timeout to Cirrus tool general tests (#58649)

Piinks committed:
Removing flag for final migration step (#58829)

Commit:
[flutter_tools] create NotifyingLogger at the top level when running flutter run --machine or flutter attach --machine (#59087)

Removes dependency on injecting additional logger with zones

engine-flutter-autoroll committed:
Roll Engine from 6fd335671508 to e8c13aa012c9 (10 revisions) (#59095)

JoseAlba committed:
Updated Slider test  (#58630)

Commit:
[flutter_tools] remove globals from proxy validator (#59175)

Remove global Platfrom from proxy validator. move tests to new file, and update asserts to cover message contents.

Commit:
[flutter_tools] inject output preferences at the top level (#58875)

The tool was setting the output preferences in a sub-context. Originally these were not injected before the arg parsers were created, though that was fixed by the lazy command creation. Once local engine is removed, the inner flutter_command Zone can be removed.

natebosch committed:
Drop an unnecessary factory constructor (#58723)

HansMuller committed:
Update TextTheme.button.letterSpacing from 0.75 to 1.25 per spec (#59008)

darrenaustin committed:
Minor cleanup of some no longer needed code. (#59111)

darrenaustin committed:
Make the InkResponse's focus highlight honor the radius parameter (#59117)

jjs1015 committed:
Fix typo in scroll_aware_image_provider.dart (#59128)

jmagman committed:
Revert "[flutter_tools] inject output preferences at the top level (#58875)" (#59197)

This reverts commit dca6320fe4455e72ef997e447146089696011cfc.

CareF committed:
Line break for devicelab/bin/run.dart help info (#58986)

The original help info has lines too long for
appropriate terminal display.

jmagman committed:
Move iOS Podfile logic into tool (#59044)

perclasson committed:
Rebuild SliverAppBar when forceElevated changes (#59162)

giantsol committed:
Add sample code to PositionedTransition (#59156)

CareF committed:
Cleanup devicelab framework duplicate (#59046)

* combine forwardStandardStreams

* combine exec and eval

blaugold committed:
Fix docs for Focus.onKey event propagation (#59096)

darrenaustin committed:
Typo fixing sweep through packages/flutter. (#59219)

a14n committed:
Opt out nnbd in packages/flutter (#59186)

* add language version 2.8 in packages/flutter

* enable non-nullable analyzer flag

engine-flutter-autoroll committed:
Roll Engine from e8c13aa012c9 to 965fbbed1776 (42 revisions) (#59222)

a14n committed:
fix tree (#59257)

wytesk133 committed:
[Flutter Driver] Update the comments regarding the default timeout of WaitFor and WaitForAbsent commands (#58771)

grouma committed:
Do not depend on embedded (#59210)

liyuqian commented:
> > Looks like this is the framework counterpart to [flutter/engine#18814](https://github.com/flutter/engine/pull/18814)? /cc @liyuqian
> > This PR is also missing tests. Can you please add some?
> 
> Which kind of test to add? Is there any examples?

Maybe something like the test [here][1]. @goderbauer : do you have some better examples?

[1]: https://github.com/flutter/flutter/blob/d25b9c80a4ab544ed2c667e3cf968ab36d0cfe10/packages/flutter/test/widgets/physical_model_test.dart#L15

zmtzawqlp committed:
add missing arguments for all constructors of ListView and GridView (#58746)

jmagman committed:
Deprecate make-host-app-editable (#59217)

AyushBherwani1998 committed:
adds interactive example (#59251)

jmagman committed:
Support .flutter-plugins-dependencies (#59209)

Commit:
Remove use of BundleUtilities in Linux build (#59080)

* Explicitly install the Flutter library

* Publish ${FLUTTER_LIBRARY} to parent scope

* Bundle plugin libraries

* Bundle <plugin_name>_bundled_libraries too

* BUNDLED_LIBRARIES -> PLUGIN_BUNDLED_LIBRARIES

* Update Linux plugin test

* Address review comments

a14n committed:
test flutter framework with null-safety (#59280)

Commit:
[versions] Update all the versions (#59283)

engine-flutter-autoroll committed:
Roll Engine from 965fbbed1776 to d417772d7acd (21 revisions) (#59282)

zanderso committed:
Revert "Roll Engine from 965fbbed1776 to d417772d7acd (21 revisions) (#59282)" (#59334)

This reverts commit 1d407de925c395e8e4e8712fb2f4f90712702600.

Piinks committed:
Support floating the header slivers of a NestedScrollView (#59187)

johnsonmh committed:
[Widgets] Add DefaultTextHeightBehavior inherited widget. (#59196)

DanTup committed:
Don't crash on requests for invalid package URLs (#59250)

jmagman committed:
Revert "[Widgets] Add DefaultTextHeightBehavior inherited widget. (#59196)" (#59347)

This reverts commit 3efc517ab5bb0263c010fc6710d2b1c32b048b2e.

Hixie committed:
Build routes even less. (#58202)

JuYeong0413 committed:
fix typo in bottom navigation bar docs (#58780)

tvolkert committed:
Add support for horizontal and vertical double-arrow system cursors (#59342)

Piinks committed:
Step 3 of 3: Remove opt-in for debugCheckHasMaterialLocalizations assertion on TextField (#58831)

jmagman committed:
Add Podfile migration warning to support federated plugins (#59201)

darrenaustin committed:
First pass at keyboard navigation for the Material Date Picker (#59279)

chaselatta committed:
Remove Fuchsia BUILD.gn files (#59285)

stuartmorgan-g committed:
CMake fix for projects without plugins (#59343)

This install step fails if the list is empty, which is the case for projects without plugins (or Dart-only plugins, so flutter/plugins presubmits are currently broken).

zljj0818 committed:
add test for Texture

zljj0818 commented:
> Looks like this is the framework counterpart to [flutter/engine#18814](https://github.com/flutter/engine/pull/18814)? /cc @liyuqian
> 
> This PR is also missing tests. Can you please add some?

I had add test for Texture widget, please review:)

zljj0818 committed:
fix spelling mistake

engine-flutter-autoroll committed:
Roll Engine from 965fbbed1776 to 801559ac4ed3 (50 revisions) (#59384)

* 141ee78b8 Roll Dart SDK from 2b917f5b6a0e to 0fab083e1c2b (15 revisions) (flutter/engine#18974)

* f5ab179e7 Call Shell::NotifyLowMemoryWarning on Android Trim and LowMemory events (flutter/engine#18979)

* 4dabac999 Roll Fuchsia Linux SDK from 8AiUM... to ThzHh... (flutter/engine#18976)

* 50cae024a Reland: Add RAII wrapper for EGLSurface (flutter/engine#18977)

* db7f226c5 Include the memory header as unique_ptr is used in ASCII Trie. (flutter/engine#18978)

* b19a17d5d Implement an EGL resource context for the Linux shell. (flutter/engine#18918)

* ca26b7558 Make Linux shell plugin constructor descriptions consistent (flutter/engine#18940)

* efd1452a4 Roll Skia from 9c401e7e1ace to f69e40841eb9 (11 revisions) (flutter/engine#18983)

* e5845af42 Put JNI functions under an interface (flutter/engine#18903)

* 1ddc6e1d3 Call destructor and fix check (flutter/engine#18985)

* 7f71f1d90 Roll Fuchsia Mac SDK from oWhyp... to gAD3P... (flutter/engine#18982)

* 74291b73b Roll Dart SDK from 0fab083e1c2b to e62b89c56d01 (10 revisions) (flutter/engine#18987)

* 2739bbfcf [web] Provide a hook to disable location strategy (flutter/engine#18969)

* 336d000b4 Roll Skia from f69e40841eb9 to 553496b66f12 (2 revisions) (flutter/engine#18992)

* b82769bdf Roll Skia from 553496b66f12 to 0ad37b87549b (2 revisions) (flutter/engine#18993)

* f5ca58b18 Roll Dart SDK from e62b89c56d01 to b0d2b97d2cd7 (4 revisions) (flutter/engine#18994)

* 2a82a0832 [dart] Account for compiler api change (flutter/engine#19002)

* 369e0a9b9 Add ui_benchmarks (flutter/engine#18945)

* a0a92d6c6 Roll Skia from 0ad37b87549b to de175abede4d (1 revision) (flutter/engine#18995)

* cea5a9c7a Roll Dart SDK from b0d2b97d2cd7 to f043f9e5f6ea (6 revisions) (flutter/engine#18997)

* d417772d7 Roll Fuchsia Mac SDK from gAD3P... to Wj0yo... (flutter/engine#19001)

* 45386229e Roll Skia from de175abede4d to 32d5cfa1f35e (15 revisions) (flutter/engine#19005)

* 71fce02dd Fix shift-tab not working by adding more GTK->GLFW key mappings. (flutter/engine#18988)

* 5ddc122f0 Fix inverted check in creating resource surface (flutter/engine#18989)

* 87d888804 Show warning if method response errors occur and error not handled. (flutter/engine#18946)

* 5171fbdc7 Roll Skia from 32d5cfa1f35e to 21bbfc6c2dfe (5 revisions) (flutter/engine#19006)

* 4bd6aea07 Always send key events, even if they're used for text input. (flutter/engine#18991)

* 9f898e98c Don't process key events when the text input is not requested (flutter/engine#18990)

* 48d709164 Roll buildroot for Windows warning update (flutter/engine#19000)

* 5f37029a2 Roll Dart SDK from f043f9e5f6ea to f0ea02bc51f8 (22 revisions) (flutter/engine#19010)

* ac809b48c onBeginFrame JNI (flutter/engine#18866)

* e1c622b4a Make SKSL caching test work on Fuchsia on arm64 (flutter/engine#18572)

* 2651beba0 Exit before pushing a trace event when layer tree holder is empty (flutter/engine#19008)

* acf048bd0 Remove log added for local testing (flutter/engine#19012)

* 1f3aa23e2 Roll Dart SDK from f0ea02bc51f8 to 0b64f5488965 (9 revisions) (flutter/engine#19013)

* 8dcc95dbd Roll Fuchsia Mac SDK from Wj0yo... to gR0Zc... (flutter/engine#19015)

* f6455fa6f Roll Dart SDK from 0b64f5488965 to 50836c171e91 (4 revisions) (flutter/engine#19017)

* b2fea9dcf Roll Skia from 21bbfc6c2dfe to 30212b7941d6 (6 revisions) (flutter/engine#19009)

* 006564625 Roll Skia from 30212b7941d6 to 3d6bf04366f6 (17 revisions) (flutter/engine#19020)

* 0a852d8ad Revert "Call Shell::NotifyLowMemoryWarning on Android Trim and LowMemory events (#18979)" (flutter/engine#19023)

* 194acdfae apply null safety syntax to mobile dart:ui (flutter/engine#18933)

* b0a0e0e32 Roll Skia from 3d6bf04366f6 to 637838d20abd (2 revisions) (flutter/engine#19021)

* be499ab46 Roll Fuchsia Mac SDK from gR0Zc... to H-uAk... (flutter/engine#19022)

* 8c24c41a0 Roll Skia from 637838d20abd to ac16760df463 (1 revision) (flutter/engine#19025)

* 7cb7003a1 onEndFrame JNI (flutter/engine#18867)

* e3fdb2330 [fuchsia] Add ability to configure separate data and asset dirs (flutter/engine#18858)

* 983b6e198 Call Shell::NotifyLowMemory when backgrounded/memory pressure occurs on Android (flutter/engine#19026)

* f7d241fd8 Wire up channel for restoration data (flutter/engine#18042)

* e84d497b9 Fix hit testing logic in fuchsia a11y (flutter/engine#19029)

* 801559ac4 Revert to last-known-good-rev of Dart SDK (flutter/engine#19031)

Hixie committed:
Remove license statements in template files. (#57268)

anisalibegic committed:
Add shadowColor to AppBar and AppBarTheme (#58708)

jamesderlin committed:
[flutter tools] Don't return success if we trigger runZoned's error callback (#58474)

duzenko committed:
LG debugging/logcat fixed (#58670)

justinmc committed:
Modernize selection menu appearance (#59115)

Hixie committed:
More useful error messages when you use Stack without a textDirection. (#59360)

Commit:
[flutter_tools] move mingit path addition back to flutter.bat (#59369)

fixes #57802

For users without git installed, flutter on windows bundles mingit. This needs to be added to the PATH in the main script, not the executed script, so that it is available to the tool

jason-simmons committed:
Revert "Prevent WhitelistingTextInputFormatter to return a empty string if the current value does not satisfy the formatter (#57264)" (#59221)

justinmc committed:
Add instructions for updating a localized string (#59273)

* Add instructions for updating a localized string

* gem_missing_localizations does nothing when I test this out, seems to not be required

* Explain generation of new locale strings

justinmc committed:
Characters package (#59267)

jmagman committed:
Remove flutter_goldens_client package dependency from tool (#59365)

dnfield committed:
Revert "Roll Engine from 965fbbed1776 to 801559ac4ed3 (50 revisions) (#59384)" (#59490)

This reverts commit e02265dd68dffb2b0db61de1459374c3b9d6ae9f.

godofredoc committed:
Update recipes location. (#59500)

The flutter infra recipes got moved to a new location and we need to
update docs with links to the new locations.

Bug:
  https://github.com/flutter/flutter/issues/45630

zanderso committed:
Revert "Modernize selection menu appearance (#59115)" (#59561)

This reverts commit b3c237ce5465a8befbf0814d36b7013e34c2abe5.

goderbauer committed:
Replace collection's SetEquals with flutter's own (#59514)

goderbauer committed:
Word substitutions (#59484)

* Word substitutions

* ++

Commit:
[flutter_tools] ensure generated entrypoint matches test and web entrypoint language version (#59291)

Ensure that the language version of the test/web generated entrypoint matches the language version of the test file to run, or the overall package language version if no annotation is provided.

GaryQian committed:
Docs 'that that' fix (#50236)

GaryQian committed:
Docs 'that that' fix (#50237)

gspencergoog committed:
Remove DiagnosticableMixin in favor of Diagnosticable (#58635)

darrenaustin committed:
Removed an unused static local key from ScrollAction. (#59436)

christopherfujino committed:
[flutter_tools] Update roll_dev.dart (#59215)

GaryQian committed:
Handle backspace edgecase in trailing whitespace formatter. (#59379)

micimize committed:
Add link to ListTile replacement guide (#59474)

jmagman committed:
Remove last references to ideviceinstaller (#59508)

justinmc committed:
InteractiveViewer mouse scale bug (#59478)

liyuqian commented on packages/flutter/lib/src/rendering/layer.dart:
```diff
@@ -603,6 +604,13 @@ class TextureLayer extends Layer {
   /// un-freezes it when it is certain that a frame with the new size is ready.
   final bool freeze;
 
+  /// Used to set the filterQuality of the texture.
```
`Used to set the filterQuality of the texture.` doesn't provide much more information than its name `TextureLayer.filterQuality`. Maybe `The quality of sampling the texture and rendering it on screen.` is more informative. The same probably also applies to the RenderObject and the Widget.

AyushBherwani1998 committed:
[MergeableMaterial] adds dividerColor property (#59481)

renyou committed:
Revert "Build routes even less. (#58202)" (#59503)

liyuqian commented on packages/flutter/lib/src/rendering/layer.dart:
```diff
@@ -603,6 +604,13 @@ class TextureLayer extends Layer {
+  /// Used to set the filterQuality of the texture.
+  ///
+  /// Use the [FilterQuality.low] quality setting to scale the texture, which corresponds to
+  /// bilinear interpolation, rather than the default [FilterQuality.none] which corresponds
```
I think this PR is going to change the default quality to `FilterQuality.low` instead of `FilterQuality.none`? Also try the Dart doc template to avoid duplicates: https://github.com/dart-lang/dartdoc#macros

liyuqian commented on packages/flutter/test/widgets/texture_test.dart:
```diff
@@ -0,0 +1,44 @@
+import 'package:flutter/widgets.dart';
+
+void main() {
+  testWidgets('Texture with default FilterQuality', (WidgetTester tester) async {
```
Maybe also add tests to ensure that the filter quality is sent to the RenderObject/Layer from the Widget.

zljj0818 replied:
OK, i will add tests after https://github.com/flutter/engine/pull/18814 is merged

zljj0818 replied:
Add TextureBox and TextureLayer related tests.

ddomingues committed:
[flutter_tools] fix the event attribute used to skip waiting (#59568)

Commit:
Temporarily disable codecov upload (#59588)

This will need to be moved, so we can skip for now to avoid issues around double uploading or changing the token config. The tool coverage is fairly steady and we're not losing any historical data.

johnsonmh committed:
Re-land "[Widgets] Add DefaultTextHeightBehavior inherited widget." (#59350)

Commit:
[flutter_tools] deprecate build aot (#59487)

This command was previously used by the re-entrant build scripts in xcode_backend.sh and build.gradle. These have since been refactored to use flutter assemble.

Deprecation the command in preparation for removal in a future release of flutter. The only current use is a test on HHH

kenzieschmoll committed:
Cancel tooltip show timer when state is deactivated (#57244)

stuartmorgan-g committed:
Switch Linux to the GTK embedding (#59287)

Updates the tooling to use the GTK embedding, rather than the GLFW embedding:
- Adds new requirements to `doctor`
- Updates the app and plugin templates to make GTK-based runners and plugins
- Stops downloading and installing the GLFW artifacts

Final part of #54860, other than cleanup.

perclasson committed:
[flutter_tools] For l10n with deferred loading, use loadLibrary for non-web too (#59539)

After Dart VM change we are now required to use loadLibrary on an import whenever it is imported as deferred.

See: https://dart-review.googlesource.com/c/sdk/+/149613

gspencergoog committed:
Remove dependency on package:collection by moving mergeSort into foundation/collections.dart (#59521)

This removes a dependency from Flutter (package:collection) by copying the implementation of mergeSort into Flutter's foundation/collections.dart.

Also, removed a reference to UnmodifiableSetView from the shortcuts code by just returning a copy instead.

justinmc committed:
Characters Package (#53381)

AlexV525 committed:
[PageTransitionsBuilder] Fix 'ZoomPageTransition' built more than once (#58686)

AyushBherwani1998 committed:
[ExpansionPanelList] adds dividerColor property (#59641)

renyou committed:
Revert "Characters Package (#53381)" (#59677)

This reverts commit e0ed12c73a3df00b72af6e5657d35035a76c90db.

engine-flutter-autoroll committed:
Roll Engine from 965fbbed1776 to 237b5f32eff8 (95 revisions) (#59681)

a14n committed:
update _isolates_io.dart for better nnbd migration (#59666)

cbracken committed:
Revert "Roll Engine from 965fbbed1776 to 237b5f32eff8 (95 revisions) (#59681)" (#59692)

Broke iOS devicelab tests.

This reverts commit 90fb2e80883c73b6de3e77c1d9863a6cd652088c.

jmagman committed:
Don't crash when pubspec isn't a map (#59632)

Commit:
[flutter_tools] handle NPE in list views method(#59626)

Fixes #59608

The VmService getter can be null, handle that in the listViews method. NNBD when

liyuqian committed:
Reland non-breaking "Add clipBehavior to widgets with clipRect #55977" (#59364)

* Revert "Revert "Add clipBehavior to widgets with clipRect (#55977)" (#58344)"

This reverts commit 1d395c5e187370f2838bb043d3b438029b484bfc.

* Add missed Overflow

CareF committed:
Add comments to flutter_driver for timeline class (#58823)

* Add comments to flutter_driver for timeline class

Mostly about explanation for the json items from
timeline.dart and timeline_summary.dart.

perclasson committed:
Set upper limit on text scaling for AppBar.title (#58094)

Commit:
[flutter_tools] make expando on vm service null safe to handle web stuff (#59624)

christopherfujino committed:
skip fuchsia_precache-linux on release branches (#59705)

chunhtai committed:
fix the widget span layout when text scale factor != 1 (#59711)

bkonyi committed:
Manual engine roll to update format of `compileExpression` RPC response (#59717)

cbracken committed:
Revert "Manual engine roll to update format of `compileExpression` RPC response (#59717)" (#59774)

Failing dev/integration_tests/ui/test_driver/keyboard_resize_test.dart
Also timing out on new_gallery_ios__transition_perf

This reverts commit c7a37c72c2d4f74faac718abf7599b1f8d59aaa9.

Commit:
[flutter_tools] add toggle `b` and service extension to change platform brightness (#59571)

A frequent request from the last Flutter developer survey was for an easier method of testing light/dark mode changes. Currently, a user needs to manually change the theme settings or adjust phone settings to see the difference. Instead we should add a toggle from the CLI, and eventually devtools/Intellij/Vscode that allows developers to override the current setting.

Fixes #59495

Adds flutter.ext.brightnessOverride service protocol which either queries the current platform brightness, or overrides it to a new value. This accepts either Brightness.light or Brightness.dark as a value.

Adds a CLI toggle b which allows the setting to be toggled manually.

Requires an update to the MediaQuery, to conditionally use a debug override when not in release mode

Commit:
[flutter_tools] remove globals from compilers (#59184)

Refactors KernelCompiler and ResidentCompiler to no longer use globals (except as a fallback for g3 migration). Improves the compilation error when running flutter test on a package without a flutter_test dependency.

Updates machine mode to output trace text to stderr

pasindud committed:
Add the ability to ignore lines depending on comments (#58656)

https://github.com/dart-lang/coverage - 0.14.0 added the ability to be able to ignore coverage depending on comments.

// coverage:ignore-line to ignore one line.
// coverage:ignore-start and // coverage:ignore-end to ignore range of lines inclusive.
// coverage:ignore-file to ignore the whole file.

Commit:
[devicelab] fix concurrent hot reload test: stderr != failure  (#59784)

The tool needs to start outputing trace text to stderr to avoid breaking machine mode. This test is bogus, and should use the exit code.

Commit:
[flutter_tools] make parent logger optional (#59786)

The addition of the non-optional parent logger broke the google3 roll. Make it optional so we can roll

JoseAlba committed:
Chips text scaling (#57745)

darrenaustin committed:
Dismiss modal routes with a keyboard shortcut (#59310)

Commit:
[flutter_tools] add missing null-safety flags (#59773)

Some of the null-safety commands were missing - plumb them through. Ensure that verbose mode shows their output, and clean up the messaging around sound-null-safety.

Fixes #59769

Adds a test that validate each of the null safety supporting build commands has everything plumbed through.

gspencergoog committed:
Make flutter and dart scripts invoke their batch file equivalents on Windows (#59789)

This makes the flutter and dart scripts invoke their batch file equivalents if running under MINGW (i.e. git-bash) on Windows.

This allows for proper locking, and makes sure that people aren't using two different (and non-mutally-aware) locking systems when running flutter on Windows.

I also fixed a couple of places where we look for MINGW32, which fails under MINGW64. It just looks for MINGW now.

goderbauer committed:
Fix doc for DecoratedBox (#59791)

stuartmorgan-g committed:
Specify encoding for vswhere output (#59607)

On Windows, Process.run assumes the output uses the system codepage by default. This allows specifying it in our wrapper, and sets the encoding for vswhere to UTF-8 since we're passing a flag that forces it to use UTF-8 output.

Fixes #53515

igorakkerman committed:
flutter.gradle: collect list of Android plugins from .flutter-plugins-dependencies (#59294)

Commit:
Revert "flutter.gradle: collect list of Android plugins from .flutter-plugins-dependencies (#59294)" (#59810)

This reverts commit 947b48e062b6f2f00aa2f2b995677c0e101ab12d.

jamesderlin committed:
[flutter tools] Change the desktop device names and IDs (#58812)

In google3, the Linux device is always available, and it has confused
people who run the Flutter doctor and see
"• Linux • Linux • linux-x64 • Linux" listed.

Rename the Linux device name to "Linux desktop" and the device ID to
be "linux". Make similar changes to the Windows and macOS
devices for consistency.  This is also  consistent with the web
devices.

The device ID change shouldn't be break -d usage since that does a
case-insensitive prefix match.

cbracken committed:
Roll the engine from 965fbbe to b5f5e63 (#59804)

This rolls the engine from:
965fbbed1776545ee681cb57f3fb1d0c2000bbcb to
b5f5e6332cb4987e9e38ffaa267733ec0a8705ba. A Dart SDK roll in this range
changed the format of the compileExpression RPC VMService endpoint,
necessitating a tool patch.

robert-ancell committed:
Use a HeaderBar for Linux applications. (#59714)

justinmc committed:
Reland modernize selection menu appearance (#59617)

jmagman committed:
Clean up PollingDeviceDiscovery dispose (#59709)

robert-ancell committed:
Fix Linux shell window default size (#59630)

The previous call set the minimum size, not just the initial size.

Commit:
Revert "Add the ability to ignore lines depending on comments (#58656)" (#59813)

This reverts commit df02518dc98bd5109dd4a3473884fbce0fc8b837.

Hixie committed:
Deprecate WhitelistingTextInputFormatter and BlacklistingTextInputFormatter (#59120)

zljj0818 committed:
add more tests

zljj0818 commented:
../../packages/flutter/lib/src/rendering/layer.dart:619:7: Error: No named parameter with the name 'filterQuality'.
      filterQuality: filterQuality,
      ^^^^^^^^^^^^^

It seems engine code not contains https://github.com/flutter/engine/pull/18814 ?

liyuqian commented:
You'll have to pull new changes from the master branch to include engine roll https://github.com/flutter/flutter/pull/59804, and rebase against it.

christopherfujino committed:
Revert "Deprecate WhitelistingTextInputFormatter and BlacklistingTextInputFormatter (#59120)" (#59870)

This reverts commit 8665e13801fe6b53a0e67866d2ee6cd5ef2083f4.

christopherfujino committed:
Re-land "Deprecate WhitelistingTextInputFormatter and BlacklistingTextInputFormatter (#59120)" (#59876)

This relands #59120, which was reverted in #59870.

yongjhih committed:
Implement Comparable<TimeOfDay> (#59317)

jason-simmons committed:
Fix the paste button label in the new version of the filtered text pasting test (#59865)

goderbauer committed:
More word substiturions (#59497)

dkwingsmt committed:
Add benchmark for Mouse region (web) (#59803)

justinmc committed:
Label unnecessarily ellided (#59807)

helin24 committed:
Add integration tests for structured error (#59809)

engine-flutter-autoroll committed:
Roll Engine from b5f5e6332cb4 to f1355815488f (5 revisions) (#59818)

Commit:
[versions] update all versions (#59832)

Fixes #59718

Update all versions, specifically dwds which has a hot restart fix

Commit:
Enabled expression evaluation by default (#59826)

srawlins committed:
Remove unused import which shares prefix name with a used import. (#59160)

Due to a bug in analyzer [1], some unused imports are not reported when
multiple import directives share a prefix name, and one or more are unused.

[1] https://github.com/dart-lang/sdk/issues/38784

engine-flutter-autoroll committed:
Roll Engine from f1355815488f to 676cd566f731 (7 revisions) (#59892)

ferhatb committed:
Fix outline button solid path when BorderSize.width is used (#51581)

zljj0818 commented:
> You'll have to pull new changes from the master branch to include engine roll #59804, and rebase against it.

May i have made a wrong update operation?

zljj0818 committed:
fix analyze errors

zljj0818 commented:
Sorry, close this pr for wrong update operation:(
